R/client.R

```diff
@@ -34,17 +34,20 @@ AmericanSoccerAnalysis <- R6::R6Class("AmericanSoccerAnalysis",
         #' @field httr_configs Configs to pass on to all \code{httr} functions. See \href{https://www.rdocumentation.org/packages/httr/versions/1.4.2/topics/config}{documentation}.
         httr_configs = list(),
 
+        #' @field latest_update_timestamp Keeps record of the latest update timestamp for queried data. Used to conditionally clear the client-side cache.
+        latest_update_timestamp = NULL,
+
         #' @description Creates a new \code{AmericanSoccerAnalysis} object.
         #' @param ... Configs to pass on to all \code{httr} functions. See \href{https://www.rdocumentation.org/packages/httr/versions/1.4.2/topics/config}{documentation}.
         #' @return A new \code{AmericanSoccerAnalysis} object.
         initialize = function(...) {
             self$base_url <- glue::glue("https://app.americansocceranalysis.com/api/{self$API_VERSION}")
             self$httr_configs <- list(...)
-            self$players <- get_entity(self, "player")
-            self$teams <- get_entity(self, "team")
-            self$stadia <- get_entity(self, "stadium")
-            self$managers <- get_entity(self, "manager")
-            self$referees <- get_entity(self, "referee")
+
+            if (!rlang::is_interactive() && !isTRUE(as.logical(Sys.getenv("NOT_CRAN", "false")))) {
+                self$latest_update_timestamp <- .get_latest_update_timestamp(self)
+                .initialize_entities(self, verbose = TRUE)
+            }
```

Comment:
This should prevent any API calls when the class is instantiated on CRAN. All other tests involving API calls are skipped as well. The only tests run on their side should be validating the `base_url` and `httr_configs` attributes. Hopefully, that means ~fewer~ no more snarky emails.